app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -167,6 +169,18 @@ def get_managed_images
         process_collection(images, :vms) { |image| parse_managed_image(image) }
       end
 
+      def get_market_images
+        urns   = @options.market_image_urns
+        images = if urns
+                   gather_data_for_this_region(@vmis).select do |image|
+                     urns.include?(image.id)
+                   end
+                 else
+                   gather_data_for_this_region(@vmis)
+                 end
```

Comment:
@01100010011001010110010101110000 - yea I realize that we can filter on URN, I'm just concerned we could end up gathering a ton of images, the vast majority of which would never be used in provisioning, and it could become a performance issue.

@blomquisg - Do you think filtering the gathering of marketplace images by URN is enough to address a possible performance issue?